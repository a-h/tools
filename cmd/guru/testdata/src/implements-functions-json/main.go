package main

// Tests of 'implements' query, -output=json.
// See go.tools/guru/guru_test.go for explanation.
// See main.golden for expected query results.

import _ "lib"

func main() {
}

type FTypeA func(s string)

func ftypea(s string) {} // @implements FTypeA "ftypea"
// guru implements ../guru/testdata/src/implements-functions/main.go:#247

type FTypeB func(i int) // @implements FTypeB "FTypeB"
// guru implements ../guru/testdata/src/implements-functions/main.go:#377

func ftypeb(i int) {}
