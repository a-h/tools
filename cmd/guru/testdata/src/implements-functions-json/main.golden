-------- @implements FTypeA --------
{
	"type": {
		"name": "implements-functions-json.ftypea",
		"pos": "testdata/src/implements-functions-json/main.go:14:6",
		"kind": "func"
	},
	"to": [
		{
			"name": "implements-functions-json.FTypeA",
			"pos": "testdata/src/implements-functions-json/main.go:12:6",
			"kind": "signature"
		}
	]
}
-------- @implements FTypeB --------
{
	"type": {
		"name": "implements-functions-json.FTypeB",
		"pos": "testdata/src/implements-functions-json/main.go:17:6",
		"kind": "signature"
	},
	"to": [
		{
			"name": "implements-functions-json.ftypeb",
			"pos": "testdata/src/implements-functions-json/main.go:20:6",
			"kind": "func"
		}
	]
}
